src/components/RichTextEditor/EditorPositioningWrapper/ToolbarWrapper/ToolbarWrapperPositioningFloating.tsx

```diff
@@ -24,14 +25,33 @@ export const ToolbarWrapperPositioningFloating = ({
         }
     }, [editorWidth, toolbarWidth, toolbarButtonGroups]);
 
+    useEffect(() => {
+        document.addEventListener('selectionchange', () => {
+            const selection = window.getSelection();
+            const selectionFrom = selection?.getRangeAt(0).startOffset;
+            const selectionTo = selection?.getRangeAt(0).endOffset;
+            const hasActiveSelection = !!selection && selectionFrom !== selectionTo;
+
+            if (!hasActiveSelection && cleanupFunction.current) {
+                cleanupFunction.current();
+            }
+        });
+    }, []);
```

Comment:
```suggestion
    useEffect(() => {
        const handleSelectionChange  = () => {
            const selection = window.getSelection();
            const selectionFrom = selection?.getRangeAt(0).startOffset;
            const selectionTo = selection?.getRangeAt(0).endOffset;
            const hasActiveSelection = !!selection && selectionFrom !== selectionTo;

            if (!hasActiveSelection && cleanupFunction.current) {
                cleanupFunction.current();
            }
        }
        document.addEventListener('selectionchange', handleSelectionChange);
         
        return () => {
            document.removeEventListener('selectionchange', handleSelectionChange);
        }
    }, []);
```

Cleanup function was missing for removing the event listener